tests/API/Management/UsersTest.php

```diff
@@ -4,61 +4,205 @@
 use Auth0\SDK\API\Management;
 use Auth0\Tests\API\BasicCrudTest;
 
-class UsersTest extends BasicCrudTest {
+/**
+ * Class UsersTest.
+ *
+ * @package Auth0\Tests\API\Management
+ */
+class UsersTest extends BasicCrudTest
+{
+    /**
+     * Unique identifier name for Users.
+     *
+     * @var string
+     */
+    protected $id_name = 'user_id';
 
-    protected $email;
+    /**
+     * Random number used for unique testing names.
+     *
+     * @var integer
+     */
+    protected $rand;
+
+    /**
+     * Should the results returned by getAll be searched for the created user?
+     *
+     * @var bool
+     */
     protected $findCreatedItem = false;
 
-    protected function getId($entity) {
-        return $entity['user_id'];
+    /**
+     * Connection to create the user in.
+     *
+     * @var string
+     */
+    protected $connection = 'Username-Password-Authentication';
+
+    /**
+     * UsersTest constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->rand = rand();
     }
 
-    protected function getApiClient($scopes = null) {
+    /**
+     * Return the Users API to test.
+     *
+     * @return Management\Users
+     */
+    protected function getApiClient()
+    {
+        $token = $this->getToken($this->env, ['users' => ['actions' => ['create', 'read', 'delete', 'update']]]);
+        $api = new Management($token, $this->domain);
+        return $api->users;
+    }
 
-        if ($scopes === null) {
-            $scopes = ['create', 'read', 'delete', 'update'];
-        }
+    /**
+     * Get the User create data to send with the test create call.
+     *
+     * @return array
+     */
+    protected function getCreateBody()
+    {
+        return [
+            'connection' => $this->connection,
+            'email' => 'test-create-user-' . $this->rand . '@auth0.com',
+            'password' => 'Y6t82hQjpXCMd3oD7Zsc',
+            'picture' => 'https://cdn.auth0.com/styleguide/components/1.0.8/media/logos/img/badge.png',
+            'email_verified' => false,
+            'user_metadata' => [
+                'key1' => 'value1',
+                'key2' => 'value2',
+            ],
+        ];
+    }
 
-        $env = $this->getEnv();
-        $token = $this->getToken($env, [
-            'users' => [
-                'actions' => $scopes
-            ]
-        ]);
+    /**
+     * Tests the \Auth0\SDK\API\Management\Users::getAll() method.
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     */
+    protected function getAllEntities()
+    {
+        $fields = array_keys($this->getCreateBody());
+        $fields[] = $this->id_name;
```

Comment:
what's this line doing? is like no index is passed to the array? 😕

Reply:
`$fields` is an array of fields to get, created initially as the keys from the `createBody` array. This adds the `id` field to that array so we get the `id` as well.

Reply:
but since "arrays in php are not arrays but dictionaries", am I right to assume this will end up adding a key named `user_id` (specified here https://github.com/auth0/auth0-PHP/pull/261/files/b018f664c75fbfaca8c8febb9148eacb2002d58f#diff-f6b74f1841de4422909e8c5982be5c87R19) with a `null` value?

Reply:
No, this is just a plain array, numeric keys:

`[ 'connection', 'email', 'password', 'picture', 'email_verified', 'user_metadata', 'user_id' ]`